doc/source/tutorial/sparse.rst

```diff
@@ -192,6 +192,30 @@ Now there are only five stored elements in our sparse array, and it is identical
           [0, 4, 1, 0],
           [0, 0, 5, 0]])
 
+Canonical formats
+-----------------
+
+Several sparse array formats have "canonical formats" to allow for more efficient operations.
+Generally these consist of added restrictions like:
+
+- No duplicate entries for any value
+- Sorted indices
+
+Classes with a canonical form include: :func:`coo_array`, :func:`csr_array`, :func:`csc_array`, and :func:`bsr_array`.
+See the docstrings of these classes for details on each canonical representation.
+
+To check if an instance of these classes is in canonical form, use the ``.has_canonical_format`` attribute:
+
+   >>> coo = sparse.coo_array(([1, 1, 1], ([0, 2, 1], [0, 1, 2])))
+   >>> coo.has_canonical_format
+   False
+
+To convert an instance to canonical form, use the ``.sum_duplicates()`` method:
+
+   >>> coo.sum_duplicates()
+   >>> coo.has_canonical_format
+   True
```

Comment:
The format in the original isn't canonical because the rows aren't sorted. It could be changed to have duplicate entries though.

I would point out the format used here is `coo`, while the additional example added in the suggestion uses `.indices`, `.indptr` instead of `.row`, `.col`

Reply:
Ah yeah, I mistyped arrays and used another revision that had this problem (this was fixed by https://github.com/scipy/scipy/pull/16936). :man_facepalming: